src/backend/distributed/planner/multi_explain.c

```diff
@@ -1489,7 +1489,8 @@ WrapQueryForExplainAnalyze(const char *queryString, TupleDesc tupleDesc)
 	 * column definition cannot be empty, so create a dummy column definition for
 	 * queries with no results.
 	 */
-	if (tupleDesc->natts == 0)
+	bool originalQueryResultHasNoColumns = (tupleDesc->natts == 0);
+	if (originalQueryResultHasNoColumns)
```

Comment:
would have a slight preference for non-negated names, i.e. `if (!originalQueryResultHasColumns)`

also, this code confused me a bit at first because originalQuery usually means the original query, and the original query did have a column (row_number). Perhaps `jobHasColumns` or `taskHasColumns`?

Also perfectly fine with using `tupleDesc->natts == 0` instead of the name.